src/WebJobs.Script.WebHost/Configuration/FlexConsumptionMetricsPublisherOptionsSetup.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    public class FlexConsumptionMetricsPublisherOptionsSetup : IConfigureOptions<FlexConsumptionMetricsPublisherOptions>
+    {
+        private IEnvironment _environment;
+
+        public FlexConsumptionMetricsPublisherOptionsSetup(IEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public void Configure(FlexConsumptionMetricsPublisherOptions options)
+        {
+            options.MetricsFilePath = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsMetricsPublishPath);
```

Comment:
The metrics file path will come as part of container start context, so this setup needs to run after specialization which I do.

Reply:
I was expecting this to be part of the initial environment in placeholder mode since we already know where this pod is hosted. 
this also works but curious why we delay the injection of this config (in the future if we ever do metrics for non placeholder based containers this would have to change so wanted to understand this better)

also it would be good if we set this up as part of initial environment before customer code and add logic to prevent customer env, from overriding this later accidentally (control plane doesnt prevent customers from setting FUNCTIONS_METRICS_PUBLISH_PATH)

Reply:
I wasn't explicitly trying to delay injection of this. Definitely if this can be established in placeholder mode, makes sense to have it then. Is there another placeholder mode setting you can point to that this should behave like (an existing example)?

Reply:
I've removed the post specialization re-initialization of this value. Therefore, because the publisher (like other metrics publishers) is a WebHost level level service created placeholder mode and the options are cached on construction, we'll be using the initial placeholder mode state.